src/uwtools/tests/drivers/test_jedi.py

```diff
@@ -0,0 +1,207 @@
+
+
+def test_JEDI_files_copied(driverobj):
+    with patch.object(jedi, "filecopy") as filecopy:
+        driverobj._driver_config["run_dir"] = "/path/to/run"
+        driverobj.files_copied()
+        assert filecopy.call_count == 2
+        assert (
+            call(src=Path("/path/to/baz"), dst=Path("/path/to/run/bar/baz"))
+            in filecopy.call_args_list
+        )
+        assert (
+            call(src=Path("/path/to/foo"), dst=Path("/path/to/run/foo")) in filecopy.call_args_list
+        )
+
+
+def test_JEDI_files_linked(driverobj):
+    with patch.object(jedi, "symlink") as symlink:
+        driverobj._driver_config["run_dir"] = "/path/to/run"
+        driverobj.files_linked()
+        assert symlink.call_count == 2
+        assert (
+            call(target=Path("/path/to/baz"), linkname=Path("/path/to/run/bar/baz"))
+            in symlink.call_args_list
+        )
+        assert (
+            call(target=Path("/path/to/foo"), linkname=Path("/path/to/run/foo"))
+            in symlink.call_args_list
+        )
+
+
+def test_JEDI_provisioned_run_directory(driverobj):
+    with patch.multiple(
+        driverobj,
+        files_copied=D,
+        files_linked=D,
+        runscript=D,
+        configuration_file=D,
+    ) as mocks:
+        driverobj.provisioned_run_directory()
+    for m in mocks:
+        mocks[m].assert_called_once_with()
+
+
+def test_JEDI_run_batch(driverobj):
+    with patch.object(driverobj, "_run_via_batch_submission") as func:
+        driverobj.run()
+    func.assert_called_once_with()
+
+
+def test_JEDI_run_local(driverobj):
+    driverobj._batch = False
+    with patch.object(driverobj, "_run_via_local_execution") as func:
+        driverobj.run()
+    func.assert_called_once_with()
+
+
+def test_JEDI_runscript(driverobj):
+    with patch.object(driverobj, "_runscript") as runscript:
+        driverobj.runscript()
+        runscript.assert_called_once()
+        args = ("envcmds", "envvars", "execution", "scheduler")
+        types = [list, dict, list, Slurm]
+        assert [type(runscript.call_args.kwargs[x]) for x in args] == types
+
+
+def test_JEDI_validate_only(caplog, driverobj):
+
+    @external
+    def file(path: Path):
+        yield "Mocked file task for %s" % path
+        yield asset(path, lambda: True)
+
+    logging.getLogger().setLevel(logging.INFO)
+    with patch.object(jedi, "file"):
+        with patch.object(jedi, "run") as run:
+            result = Mock(output="", success=True)
+            run.return_value = result
+            driverobj._driver_config["configuration_file"]["base_file"] = "/path/to/jedi.yaml"
+            driverobj.validate_only()
+            cmds = [
+                "module load some-module",
+                "module load jedi-module",
+                "time /path/to/qg_forecast.x --validate-only /path/to/jedi.yaml 2>&1",
+            ]
+            run.assert_called_once_with("20240201 18Z jedi validate_only", " && ".join(cmds))
```

Comment:
Does the string `"20240201 18Z jedi validate_only"` actually get passed as an argument to `run()`? Or is that the `iotaa` `run()` that I commented on above?

Reply:
It's the `iotaa` `run()` that does take a `taskname` argument to make its logging more explicit.

Reply:
Thanks for the clarification.